code/modules/surgery/organs_internal.dm

```diff
@@ -101,6 +101,104 @@
 		if(I && I.damage > 0)
 			I.take_damage(dam_amt,0)
 
+// Repairs mechanical organs, including augments
+/datum/surgery_step/internal/fix_mech_organ
+	allowed_tools = list(
+	/obj/item/stack/nanopaste = 100,
+	/obj/item/surgery/bonegel = 30,			// Taken with modifications from robot organ repair
+	/obj/item/screwdriver = 50				// Screwdriver nerfed, since 70% is basically free repairs
+	)
+
+	min_duration = 70
+	max_duration = 90
+
+/datum/surgery_step/internal/fix_mech_organ/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	if(!ishuman(target))
+		return
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+	if(!affected)
+		return
+	var/is_organ_damaged = 0
+	for(var/obj/item/organ/I in affected.internal_organs)
+		if(I.damage > 0 && I.robotic >= 2)
+			is_organ_damaged = 1
+			break
+	return ..() && is_organ_damaged
+
+/datum/surgery_step/internal/fix_mech_organ/begin_step(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	if(!ishuman(target))
+		return
+
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+
+	for(var/obj/item/organ/internal/I in affected.internal_organs)
+		if(I && I.damage > 0 && I.robotic >= 2)
+			user.visible_message("[user] starts mending the damage to [target]'s [I.name]'s mechanisms.", \
+					"You start mending the damage to [target]'s [I.name]'s mechanisms." )
+
+	target.custom_pain("The pain in your [affected.name] is living hell!", 75)
+	..()
+
+/datum/surgery_step/internal/fix_mech_organ/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	if(!ishuman(target))
+		return
+	var/obj/item/organ/external/affected = target.get_organ(target_zone)
+
+	for(var/obj/item/organ/internal/I in affected.internal_organs)
+		if(I && I.damage > 0 && I.robotic >= 2)
+			user.visible_message("<span class='notice'>[user] repairs [target]'s [I.name] with \the [tool].</span>", \
+					"<span class='notice'>You repair [target]'s [I.name] with \the [tool].</span>" )
+			// No surgical_fix, since we probably can't have scar tissue on eye augments.
+			// This would be realistic, but no real need here
+			I.damage = 0
```

Comment:
This is not acceptable. Would be way too OP for something that has no costs.

Reply:
Internal organ surgery has no cost either (the packs are not used up), and robotic organ repair can be done with a 70% screwdriver. This one is actually nerfed, since it's only 50% screwdriver with much more damage on failure.

Reply:
The point is that surgical_fix applies scarring and this is not negotiable - it will have to stay.